core/services/vrf/go_generate_test.go

```diff
@@ -216,5 +220,30 @@ func compileCommand(t *testing.T) string {
 		}
 		panic(err)
 	}
-	return string(cmd)
+	return strings.Trim(string(cmd), "\n")
+}
+
+func boxOutput(s string, args ...interface{}) string {
+	s = fmt.Sprintf(s, args...)
+	lines := strings.Split(s, "\n")
+	maxlen := 0
+	for _, line := range lines {
+		if len(line) > maxlen {
+			maxlen = len(line)
+		}
+	}
+	internalLength := maxlen + 4
+	rv := "↘" + strings.Repeat("↓", internalLength) + "↙\n" // top line
+	rv += "→  " + strings.Repeat(" ", maxlen) + "  ←\n"
+	readme := strings.Repeat("README ", maxlen/7)
+	rv += "→  " + readme + strings.Repeat(" ", maxlen-len(readme)) + "  ←\n"
+	rv += "→  " + strings.Repeat(" ", maxlen) + "  ←\n"
+	for _, line := range lines {
+		rv += "→  " + line + strings.Repeat(" ", maxlen-len(line)) + "  ←\n"
+	}
+	rv += "→  " + strings.Repeat(" ", maxlen) + "  ←\n"
+	rv += "→  " + readme + strings.Repeat(" ", maxlen-len(readme)) + "  ←\n"
+	rv += "→  " + strings.Repeat(" ", maxlen) + "  ←\n"
+	return "\n" + rv + "↗" + strings.Repeat("↑", internalLength) + "↖" + // bottom line
+		"\n\n"
 }
```

Comment:
This function looks a little complex, could it be tested?

Reply:
It's already a test helper to make error messages easier to spot. It can only be used in tests, because it's a `_test.go` file. Does it really need to be tested?